docs/docker-stack/build.rst

```diff
@@ -274,18 +274,27 @@ You should be aware, about a few things:
 Examples of image extending
 ---------------------------
 
-Example of upgrading Airflow Provider packages
+Example of customizing Airflow Provider packages
 ..............................................
 
 The :ref:`Airflow Providers <providers:community-maintained-providers>` are released independently of core
 Airflow and sometimes you might want to upgrade specific providers only to fix some problems or
 use features available in that provider version. Here is an example of how you can do it
 
-.. exampleinclude:: docker-examples/extending/add-providers/Dockerfile
+.. exampleinclude:: docker-examples/extending/custom-providers/Dockerfile
     :language: Dockerfile
     :start-after: [START Dockerfile]
     :end-before: [END Dockerfile]
 
+Example of adding Airflow Provider packages
```

Comment:
We already have one guide about java instalation: https://airflow.apache.org/docs/docker-stack/recipes.html#apache-hadoop-stack-installation
I am not sure if we should add new sections with the same or if it is enough to add a reference to an existing section.

Reply:
Thanks @mik-laj !

I don't mind moving it to the "Recipes" document or closing this PR if you think that it's redundant.
As a Spark user (using `SparkSubmitOperator` in AWS) I think that the example I added is short and simple. 

Seems like this one is dealing with java, Hadoop, Hive, Spark, GCP.
* I didn't understand how it installs Spark: seems like it only creates a spark directory and downloads some google cloud jars. Where are the [Spark files](https://www.apache.org/dyn/closer.lua/spark/spark-3.2.1/spark-3.2.1-bin-hadoop3.2.tgz) themselves? By using `apache-airflow-providers-apache-spark` I'm getting it automatically (from `pyspark`). What am I missing?
* In my use case, which I assume is relatively common, Hadoop/Hive/GCP are not required. Only Spark installation is required to trigger a submit and Hadoop relevant jars are on the spark driver which run on a different pod/instance.
* Java installation: isn't it better to give an example with officially debian package installation rather than downloading from a 3rd party `jfrog`?